drivers/pinctrl/pinctrl_rts5817.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2025 Realtek Semiconductor, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/rts5817_pinctrl.h>
+#include "pinctrl_rts5817.h"
+
+#define PAD_CFG_SIZE 0x40
+
+static void pinctrl_configure_pin(pinctrl_soc_pin_t pincfg)
+{
+	uint32_t base;
+	uint32_t value;
+	uint8_t h3l1 = pincfg.power_source & BIT(0);
+	uint8_t iev18 = (pincfg.power_source & BIT(1)) >> 1;
+
+	if (pincfg.pin >= RTS_FP_PIN_AL0 && pincfg.pin <= RTS_FP_PIN_AL2) {
+		base = DT_REG_ADDR_BY_IDX(DT_NODELABEL(pinctrl), 1);
+
+		value = sys_read32(base);
+
+		value &= ~((0x1 << (pincfg.pin - RTS_FP_PIN_AL0 + AL_GPIO_PU_CTRL_OFFSET)) |
+			   (0x1 << (pincfg.pin - RTS_FP_PIN_AL0 + AL_GPIO_PD_CTRL_OFFSET)) |
+			   (0x1 << (pincfg.pin - RTS_FP_PIN_AL0 + AL_GPIO_SEL_OFFSET)));
```

Comment:
Non-blocking nit - use `BIT` macro here and through the file.
```suggestion
		value &= ~(BIT(pincfg.pin - RTS_FP_PIN_AL0 + AL_GPIO_PU_CTRL_OFFSET) |
			   BIT(pincfg.pin - RTS_FP_PIN_AL0 + AL_GPIO_PD_CTRL_OFFSET) |
			   BIT(pincfg.pin - RTS_FP_PIN_AL0 + AL_GPIO_SEL_OFFSET));
```

Reply:
Modified